File: ios/NHSOnline/NHSOnline/Utils/SchemeHandlers.swift
import Foundation

class SchemeHandlers {
    private var handlers = Dictionary<String?, BaseSchemeHandler>()
    
    func registerHandler(handler: BaseSchemeHandler) {
        handlers[handler.Scheme] = handler
    }
    
    func handleUrl(url: URL) -> Bool {
        let uc = URLComponents(url: url, resolvingAgainstBaseURL: true)
        
        if let selectedHandler = handlers[uc?.scheme] {
            return selectedHandler.handle(url: url)
        }
        return false
    }
}

File: android/app/src/main/java/com/nhs/online/nhsonline/support/schemehandlers/SchemeHandlers.kt
package com.nhs.online.nhsonline.support.schemehandlers

import android.util.Log
import com.nhs.online.nhsonline.Application

class SchemeHandlers {

    private val handlers: HashMap<String, ISchemeHandler> = hashMapOf()

    fun registerHandler(handler: ISchemeHandler) {
        handlers[handler.scheme] = handler
    }

    fun handleUrl(url: String) : Boolean {
        Log.d(Application.TAG, "${this::class.java.simpleName}: Entering handleUrl")

        val handler = handlers[extractHandlerKey(url)]

        if(handler != null) return handler.handle(url)

        return false
    }

    private fun extractHandlerKey(url: String): String {
        return url.substringBefore(":") + ":"
    }
}

